Dockerfile

```diff
@@ -40,21 +40,20 @@ ENV POSTGRES_JDBC_VER=42.2.23
 COPY build.gradle settings.gradle gradlew /gradle/
 COPY gradle /gradle/gradle
 ENV GRADLE_JARS_DIR=gradle_jars
-RUN /gradle/gradlew -p /gradle build
-RUN cp -r /gradle/${GRADLE_JARS_DIR}/* /opt/bitnami/spark/jars/
+RUN /gradle/gradlew -p /gradle build && \
+    cp -r /gradle/${GRADLE_JARS_DIR}/* /opt/bitnami/spark/jars/ && \
+    rm -rf /gradle
 
 # make an empty yarn conf dir to prevent spark from complaining
 RUN mkdir -p /opt/yarn/conf && chown -R spark_user:spark /opt/yarn
 ENV YARN_CONF_DIR=/opt/yarn/conf
 
-# install pipenv
-RUN pip3 install pipenv
-
-# install python dependencies
+# Install pipenv and Python dependencies with cache cleanup
+RUN pip3 install --no-cache-dir pipenv
 COPY Pipfile* ./
-RUN pipenv sync --system
+RUN pipenv sync --system && pip cache purge
```

Comment:
Should this be `pipenv --clear` rather than pip?

Reply:
:+1:

Reply:
yup. 👍